span.go

```diff
@@ -34,6 +31,9 @@ type Span interface {
 	// Finish() should be the last call made to any span instance, and to do
 	// otherwise leads to undefined behavior.
 	Finish()
+	// FinishWithOptions is like Finish() but with explicit control over
+	// timestamps and log data.
+	FinishWithOptions(opts FinishOptions)
```

Comment:
I was naming the struct after the method, and `FinishSpan` seems redundant... thoughts?